### DISCUSSION TITLE
Multitrack MultitrackTracks type

### DISCUSSION THREAD

**Discussion by WooWan** [katspaugh/wavesurfer.js#3139] [General]
When I use multitrack.create(), the first parameter's type is an array of TrackOptions. The options within TrackOptions is of type WaveSurferOptions. 
The WaveSurferOptions type requires a container parameter. However, I believe that the container parameter may not be necessary, at least not as a required type, since we already pass the container in MultiTrackOptions (as the second parameter).

I discovered this fantastic library just a week ago—thank you for your great work! If I have misunderstood something, please let me know. I am open to creating a pull request if my suggestion is accepted.

## Minimal code snippet

```TypeScript
const multitrack = Multitrack.create(
      [
        {
          id: 0,
          url: '/music/mp3/vocal.mp3',
          volume: 0.5,
          options: {
            height: 120,
            waveColor: 'hsl(46, 87%, 49%)',
            progressColor: 'hsl(46, 87%, 20%)'
            //Should we pass container here?
          }
        },
      ],
      {
        container: playerRef.current, // required!
        minPxPerSec: 5, // zoom level
        rightButtonDrag: true, // drag tracks with the right mouse button
        cursorWidth: 2,
        cursorColor: '#D72F21',
        trackBorderColor: 'black',
        envelopeOptions: {
          lineWidth: 0,
          dragPointSize: 0
        }
      }
    );
```

## Expected result
Could we create new type using WaveSurferOptions like below?
```TypeScript
  type MultitrackWaveSurfer = Omit<WaveSurferOptions, 'container'>;
```
## Obtained result

Property 'container' is missing in type '{ height: number; waveColor: string; progressColor: string; }' but required in type 'WaveSurferOptions'.ts(2741)

## Screenshots
<img width="888" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/47740690/c4172be4-85f1-404a-917c-6b5a7fa50610">

**Comment by katspaugh** [comment]
Good catch!
It actually [overrides a bunch of options](https://github.com/katspaugh/wavesurfer-multitrack/blob/main/src/multitrack.ts#L154-L164
) when creating an internal wavesurfer instance, so they can all be omitted:

```js
    const ws = WaveSurfer.create({
      ...track.options,
      container,
      minPxPerSec: 0,
      media: this.audios[index],
      peaks: track.peaks,
      cursorColor: 'transparent',
      cursorWidth: 0,
      interact: false,
      hideScrollbar: true,
    })
```
